### PR TITLE
adding context module and fix deletion call

### PR DESCRIPTION
**Pull request by abhilashshetty04**

### CODE REVIEW

File: k8s/operators/src/pool/diskpool/context.rs [hunk 1]
```diff
@@ -0,0 +1,599 @@
...
+    /// Hashtable of name and the full last seen CRD
+    inventory: tokio::sync::RwLock<HashMap<String, ResourceContext>>,
+    /// HTTP client
+    http: clients::tower::ApiClient,
+    /// Interval
+    interval: u64,
+}
+
+impl OperatorContext {
+    pub(crate) fn new(
```
Comment (tiagolobocastro) [inline k8s/operators/src/pool/diskpool/context.rs]: missing doc comment

File: k8s/operators/src/pool/diskpool/context.rs [hunk 2]
```diff
@@ -0,0 +1,599 @@
...
+        }
+    }
+    /// Remove the resource from the operator
+    pub(crate) async fn remove(&self, name: String) -> Option<ResourceContext> {
+        let mut i = self.inventory.write().await;
+        let removed = i.remove(&name);
+        if let Some(removed) = removed {
+            info!(name =? removed.name_any(), "removed from inventory");
+            return Some(removed);
+        }
```
Comment (tiagolobocastro) [inline k8s/operators/src/pool/diskpool/context.rs]:
```suggestion
        if let Some(removed) = i.remove(&name) {
            info!(name =? removed.name_any(), "removed from inventory");
            return Some(removed);
        }
```

File: k8s/operators/src/pool/diskpool/context.rs
Comment (tiagolobocastro) [inline k8s/operators/src/pool/diskpool/context.rs:1]: hmm is this in the right place or should it be next to main.rs ? CC @Abhinandan-Purkait

**Review by tiagolobocastro — COMMENTED** [review]
chore(dsp): adding context module 
Should be refactor(dsp): ...
And
chore: do not process dsp object if removed from inventory 
Should be fix(dsp):

Also PR doesn't need "chore:"

File: k8s/operators/src/pool/diskpool/context.rs (hunk 2 above)
Comment (abhilashshetty04) [reply k8s/operators/src/pool/diskpool/context.rs]: Done

File: k8s/operators/src/pool/diskpool/context.rs
Comment (abhilashshetty04) [reply k8s/operators/src/pool/diskpool/context.rs:1]: @tiagolobocastro , As discussed, Ill move this outside diskpool.

**Review by tiagolobocastro — APPROVED** [review]

**Review by Abhinandan-Purkait — APPROVED** [review]

**Comment by abhilashshetty04** [comment]
bors merge

**Comment by abhilashshetty04** [comment]
bors cancel

**Comment by abhilashshetty04** [comment]
bors merge

**Pull request merged**